docs/design/design_trivy_file_system_scanner.md

```diff
@@ -0,0 +1,200 @@
+# Proposal: Vulnerability scanning using Trivy file system scanner
+
+Author(s): [ [Devendra Turkar](https://github.com/deven0t), [Daniel Pacak](https://github.com/danielpacak) ]
+
+## Abstract
+
+Add additional support in Trivy plugin to support file system scanning of an image. 
+
+## Background
+
+Starboard currently uses Trivy  plugin to scan and generate Vulnerability Report for an image. Trivy can scan a container 
+image by pulling the image from a remote registry. This mechanism works good for public registry image, but for private 
+registry image Trivy is dependent on registry credentials. Currently, starboard has mechanism to provide these credentials
+using ImagePullSecret or with ServiceAccount, based on how it is provided to application pod.
+With this approach we have following problem:
+1. When image pulled from private registry without ImagePullSecret or service account 
+(https://kubernetes.io/docs/concepts/containers/images/#configuring-nodes-to-authenticate-to-a-private-registry)  
+2. When image pulled from managed registry of managed cluster
+3. When pod is running with image present on host
+
+## Proposal
+
+Use Trivy file system scanning option to scan an image. So that we will not need to provide registry credentials to trivy.
+K8s scan job will use underline secrets available to pull an image and that image will be scanned by trivy as file system
```

Comment:
For the current proposal the main idea is to schedule a scan job on the same node where the scanned workload so we can leverage container image cached on that node and scan without providing credentials to Trivy.